### ISSUE TITLE
tune.exfat breaks the FS in some cases

### ISSUE THREAD

**Issue by progo** [exfatprogs/exfatprogs#202]
I have a camera that will format into exfat if the card is sufficiently large. The card formatted-in-camera will mount okay on my gentoo PC but when I try to give the card a label, the FS breaks.

Steps to reproduce
----------------------

Camera in question: Olympus OM-D E-M1 Mark 2, firmware 3.6.

1. Take a suitably large card so that the camera will default to an
   exfat formatting (64 GiB)
2. Format it in-camera (and take a snapshot on it)
3. mount it on PC.

```
      $ mount /dev/sdg1 -t exfat sdhc
      $ tree sdhc
      sdhc
      ├── ALBM
      │   └── ALB67ZZ7.BIN
      └── DCIM
          └── 100OLYMP
              └── P9120001.ORF

      3 directories, 2 files
```

4. Umount it.
5. (make an image of the block device using bzip2 at this point)
6. Give it a label using the following invocation:

```
      $ tune.exfat -L 'OLY' '/dev/sdg1'
      exfatprogs version : 1.1.3
      new label: OLY
```

7. Try mounting this using kernel exfat support

```
     $ mount /dev/sdg1 -t exfat /media/sdhc

     mount: /media/sdhc: wrong fs type, bad option, bad superblock on
     /dev/sdg1, missing codepage or helper program, or other error
```

   dmesg messages related to this incident:

```
     [Mon Sep 12 13:51:35 2022] exFAT-fs (sdg1): failed to load alloc-bitmap
     [Mon Sep 12 13:51:35 2022] exFAT-fs (sdg1): failed to recognize exfat type
```

   Trying to mount this via FUSE-exfat version 1.3.0 I get a similar
   error message.

```
     $ mount /dev/sdg1 -t exfat-fuse /media/sdhc 
     FUSE exfat 1.3.0
     ERROR: clusters bitmap is not found.
```

8. The camera doesn't recognize the FS either.
9. I formatted it on PC:

```
     $ mkfs.exfat -L OLY /dev/sdg1
     exfatprogs version : 1.1.3
     Creating exFAT filesystem(/dev/sdg1, cluster size=131072)

     Writing volume boot record: done
     Writing backup volume boot record: done
     Fat table creation: done
     Allocation bitmap creation: done
     Upcase table creation: done
     Writing root directory entry: done
     Synchronizing...

     exFAT format complete!

     $ mount /dev/sdg1 /media/sdhc
     FUSE exfat 1.3.0

     $ umount /media/sdhc
     $ 
```

   This way the card seems to be remain readable both in my system and in-camera.

The block device image
------------------
It is of a 58-gibibyte FS, 300 megabytes bzip2'd. You can find it here. https://fwwm.us/p/oly_sdcard_before_labeling.img.bz2

I made it of the first allocated partition sdg1 instead of the whole sdg. Making another will take me an hour so let me know if you need it.

Please let me know if you need something else.

**Comment by namjaejeon** [comment]
@progo Thanks for your detailed report! I have reproduced it. Could you please provide the exfat image that after setting the label on windows ? So, You can do the following step.

1. format it in your camera.
2. set the label on windows PC.
3. dump your storage and give me it.

**Comment by progo** [comment]
Alright.

1. Formatted in-camera
2. labeled the card on a Windows 10 machine "XYZZY"
3. The label isn't readable by `tune.exfat` but `mount` can still find it. Interesting! (Running a 5.17 kernel)
```
$ tune.exfat -l /dev/sdg1
exfatprogs version : 1.1.3
label: 

$ mount LABEL=XYZZY /media/sdhc
FUSE exfat 1.3.0

$ tree sdhc
sdhc
├── ALBM
│   └── ALB67ZZ7.BIN
└── System Volume Information
    ├── IndexerVolumeGuid
    └── WPSettings.dat

2 directories, 3 files
```
4. Making the block device image: https://fwwm.us/p/oly_sdcard_win10_labeled.img.bz2
5. (Curiously, this fs isn't readable by the camera! :P)

**Comment by namjaejeon** [comment]
@progo Thanks for sharing more info. I have found what is problem now. your camera tools doesn't seem to add the label entry. I just want to know how/where windows add label entry to exfat. I will check it.

**Comment by namjaejeon** [comment]
@progo Can you confirm your issue is fixed using #exfat-next branch ?

git clone --branch=exfat-next https://github.com/exfatprogs/exfatprogs

**Comment by progo** [comment]
@namjaejeon Thank for quick action. I have time to check this on Saturday (UTC+3), expect a response then. :)

**Comment by namjaejeon** [comment]
@progo Okay!

**Comment by namjaejeon** [comment]
@progo Thanks for your confirmation:).

**Comment by namjaejeon** [comment]
I have released 1.2.0 version included this issue fixes. Close this issue. Thanks.
